jdaviz/utils.py

```diff
@@ -318,57 +324,47 @@ def flux_conversion(spec, values, original_units, target_units):
     else:
         spectral_values = spec.spectral_axis
 
+    # Need this for setting the y-limits
+    eqv = u.spectral_density(spectral_values)
+
+    orig_units = u.Unit(original_units)
+    orig_bases = orig_units.bases
+    targ_units = u.Unit(target_units)
+    targ_bases = targ_units.bases
+
     # Ensure a spectrum passed through Spectral Extraction plugin
-    if '_pixel_scale_factor' in spec.meta and len(values) != 2:
+    if (('_pixel_scale_factor' in spec.meta) and
+            (((u.sr in orig_bases) and (u.sr not in targ_bases)) or
+             ((u.sr not in orig_bases) and (u.sr in targ_bases)))):
         # Data item in data collection does not update from conversion/translation.
-        # App wide original data units are used for conversion, original and
+        # App-wide original data units are used for conversion, original and
         # target_units dictate the conversion to take place.
+        n_values = len(values)
+
+        # Make sure they are float (can be Quantity).
```

Comment:
what units would they have?  Is there any chance we'd need to translate them?

Reply:
@pllim I think you will have more insight into this change

Reply:
It's a question for @astrofrog . I merely corrected the comment because it did not reflect the actual logic.

Reply:
can't we just check the code and see if it can ever be set as a quantity (and if so, its _probably_ always unitless, right?)?

Reply:
We can but we need to know the right triggers.

Reply:
we set this (`meta['_pixel_scale_factor']`) internally, I'm just asking for a case where that is set to a quantity, or if it assumes steradians -> pixels, then calling `.value` will need to eventually be changed to `to_value` for @cshanahan1's work.  

Either way, I'm hoping we can count on it either always or never being a quantity since we set it internally.

Reply:
If it is set from FITS header, it would be non-Quantity. If it is set from FITS table entry (which is theoretically possible if you read from reference file but I don't think we currently do that), it *might* be a Quantity (if there is a TUNIT that goes with the column). If it is set from ASDF, it depends on the data model and since ASDF has no standards for astronomical data, we will never know if it always is Quantity or not.